src/js/controller/model.js

```diff
@@ -233,6 +233,16 @@ define([
             this.set('renderCaptionsNatively', _provider.renderNatively);
         };
 
+        this.detachMedia = function() {
+            _provider.off('all', _videoEventHandler, this);
+            return _provider.detachMedia();
+        };
+
+        this.attachMedia = function() {
```

Comment:
You probably want to add `_provider.off('all', _videoEventHandler, this);` here to make sure never end up attaching the same listeners more than once.